### PR TITLE
Require Java 17 or newer

### PR DESCRIPTION
**Pull request by basil**
https://groups.google.com/g/jenkinsci-dev/c/gsXAqOQQEPc/m/C_EIMal6BAAJ

This cannot be merged until after we ship the June 11 weekly. It is intended for the June 18 weekly. However, the code is ready right now.

### Testing done

Tested that an appropriate error message was displayed when launching the WAR on Java 11.

### Proposed changelog entries

- Require Java 17 or newer.

### Proposed upgrade guidelines

See [this comment](https://issues.jenkins.io/browse/JENKINS-60891?focusedId=448530&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-448530).

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Review by timja — APPROVED** [review]

File: core/src/main/java/jenkins/monitor/JavaVersionRecommendationAdminMonitor.java
Comment (daniel-beck) [inline core/src/main/java/jenkins/monitor/JavaVersionRecommendationAdminMonitor.java:85]: Seems this could be removed, as whatever upcoming LTS baseline includes this change will not work with Java 11 anyway, so this is effectively dead code.

**Review by daniel-beck — APPROVED** [review]

File: core/src/main/java/jenkins/monitor/JavaVersionRecommendationAdminMonitor.java
Comment (basil) [reply core/src/main/java/jenkins/monitor/JavaVersionRecommendationAdminMonitor.java]: I don't have the brain cells left to think about this, so I will trust that your judgment is correct.

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by basil** [comment]
Added the `upgrade-guide-needed` label suggesting we include [this information](https://issues.jenkins.io/browse/JENKINS-60891?focusedId=448530&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-448530) for `envinject` plugin users.

**Pull request merged**
